src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExprFactory.cs

```diff
@@ -36,62 +36,46 @@ public ExprCall CreateCall(EXPRFLAG nFlags, CType pType, Expr pOptionalArguments
                )
               ));
 
-            ExprCall rval = new ExprCall();
-            rval.Kind = ExpressionKind.EK_CALL;
-            rval.Type = pType;
+            ExprCall rval = new ExprCall(pType);
             rval.Flags = nFlags;
             rval.OptionalArguments = pOptionalArguments;
             rval.MemberGroup = pMemberGroup;
             rval.NullableCallLiftKind = NullableCallLiftKind.NotLifted;
-            rval.CastOfNonLiftedResultToLiftedType = null;
 
             rval.MethWithInst = MWI;
-            Debug.Assert(rval != null);
```

Comment:
I removed all of the assertions of this type in this file. It's impossible for the result of a `new` to be null unless assigned null after the `new` so these assertions don't assert anything one can't already determine statically by considering just the scope in question.